java/java-common/src/main/java/org/cloudburstmc/protocol/java/JavaClient.java

```diff
@@ -0,0 +1,183 @@
+import io.netty.channel.ChannelOption;
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.epoll.Epoll;
+import io.netty.channel.epoll.EpollSocketChannel;
+import io.netty.channel.socket.SocketChannel;
+import io.netty.channel.socket.nio.NioSocketChannel;
+import io.netty.handler.timeout.ReadTimeoutHandler;
+import lombok.Getter;
+import lombok.Setter;
+import org.cloudburstmc.protocol.java.handler.JavaStatusPacketHandler;
+import org.cloudburstmc.protocol.java.pipeline.PacketDecoder;
+import org.cloudburstmc.protocol.java.pipeline.PacketEncoder;
+import org.cloudburstmc.protocol.java.pipeline.VarInt21FrameDecoder;
+import org.cloudburstmc.protocol.java.pipeline.VarInt21LengthFieldPrepender;
+import org.cloudburstmc.protocol.java.packet.State;
+import org.cloudburstmc.protocol.java.packet.handshake.HandshakingPacket;
+import org.cloudburstmc.protocol.java.packet.status.StatusRequestPacket;
+import org.cloudburstmc.protocol.java.packet.status.StatusResponsePacket;
+import org.cloudburstmc.protocol.java.packet.type.JavaPacketType;
+
+import javax.annotation.Nullable;
+import java.net.InetSocketAddress;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Consumer;
+
+@Getter
+public class JavaClient extends Java {
+    private InetSocketAddress remoteAddress;
+    private JavaClientSession session;
+    private int timeout;
+    private boolean closed = false;
+    private boolean connected = false;
+
+    @Setter private JavaEventHandler<JavaClientSession> handler;
+
+    private Bootstrap bootstrap;
+
+    public JavaClient() {
+        this(null, EventLoops.commonGroup());
+    }
+
+    public JavaClient(InetSocketAddress bindAddress) {
+        this(bindAddress, EventLoops.commonGroup());
+    }
+
+    public JavaClient(EventLoopGroup eventLoopGroup) {
+        this(null, eventLoopGroup);
+    }
+
+    public JavaClient(InetSocketAddress bindAddress, EventLoopGroup eventLoopGroup) {
+        this(bindAddress, eventLoopGroup, 30_000);
+    }
+
+    public JavaClient(EventLoopGroup eventLoopGroup, int timeout) {
+        this(null, eventLoopGroup, timeout);
+    }
+
+    public JavaClient(@Nullable InetSocketAddress bindAddress, EventLoopGroup eventLoopGroup, int timeout) {
+        super(bindAddress, eventLoopGroup);
+
+        this.timeout = timeout;
+    }
+
+    @Override
+    public InetSocketAddress getBindAddress() {
+        if (this.bindAddress == null) {
+            return (InetSocketAddress) this.session.getChannel().localAddress();
+        }
+        return super.getBindAddress();
+    }
+
+    @Override
+    protected void onTick() {
+        if (this.session.isConnected()) {
+            this.session.tick();
+        }
+    }
+
+    @Override
+    public CompletableFuture<Void> bind() {
+        Preconditions.checkNotNull(this.eventLoopGroup, "Event loop group was null");
+
+        Class<? extends SocketChannel> channel = NioSocketChannel.class;
+        if (Epoll.isAvailable()) {
```

Comment:
Protocol listens to a flag that acts as a hard disable for Epoll; we should respect that flag here as well.